<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IITAKenyaImputation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>IITAKenyaImputation/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220301code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220301)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220301code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220301)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLucianoRogerioIITAKenyaImputationtreede45af2d2d35f0ca2f76c485c57e9452d064fb48targetblankde45af2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/LucianoRogerio/IITAKenyaImputation/tree/de45af2d2d35f0ca2f76c485c57e9452d064fb48" target="_blank">de45af2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomLucianoRogerioIITAKenyaImputationtreede45af2d2d35f0ca2f76c485c57e9452d064fb48targetblankde45af2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/LucianoRogerio/IITAKenyaImputation/tree/de45af2d2d35f0ca2f76c485c57e9452d064fb48" target="_blank">de45af2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  data/.DS_Store
    Untracked:  data/BreedBaseGenotypesDownload.vcf
    Untracked:  data/DCas21_6651.samples
    Untracked:  data/DCas21_6651.sitesWithAlleles
    Untracked:  data/DCas21_6651.vcf
    Untracked:  data/DCas21_6651.vcf.gz
    Untracked:  data/Report_6651_VCF_Ref_Version6.txt
    Untracked:  data/SEQ_SNPs_counts_0_Target_extend_Ref.csv
    Untracked:  data/ServiceSpecification-DCas21-6651.pdf
    Untracked:  data/TMEB14CassavaBase.vcf
    Untracked:  data/dart_sample_trackingfile_template_MEF.csv
    Untracked:  output/DosageMatrix_DCas21_6651_WA_REFimputedAndFiltered.rds

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/LucianoRogerio/IITAKenyaImputation/blob/de45af2d2d35f0ca2f76c485c57e9452d064fb48/analysis/index.Rmd" target="_blank">de45af2</a>
</td>
<td>
LucianoRogerio
</td>
<td>
2022-03-01
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="imputation-work-for-iita-kenya-morag-ferguson" class="section level2">
<h2>Imputation work for IITA Kenya (Morag Ferguson)</h2>
<p>I followed the script wrote by <a href="https://github.com/wolfemd">Marnin Wolfe</a> on <a href="https://wolfemd.github.io/IITA_2021GS/ImputeDCas21_6038.html">IITA_2021GS repository</a>. Great Guy</p>
<pre class="bash"><code>screen -R IITAImputation
module load R/4.1.2
R</code></pre>
<div id="input-parameters" class="section level3">
<h3>Input Parameters</h3>
<pre class="r"><code>devtools::install_github(&quot;LucianoRogerio/genomicMateSelectR&quot;, ref = &#39;master&#39;) 

library(tidyverse); library(magrittr)
library(genomicMateSelectR)

dartvcfInput&lt;-here::here(&quot;data/&quot;,&quot;Report_6651_VCF_Ref_Version6.txt&quot;)
dartcountsInput&lt;-here::here(&quot;data/&quot;,&quot;SEQ_SNPs_counts_0_Target_extend_Ref.csv&quot;)
outName&lt;-here::here(&quot;output&quot;,&quot;DCas21_6651&quot;)
nskipvcf&lt;-2; nskipcounts&lt;-2; ncores&lt;-4</code></pre>
</div>
<div id="check-format-of-dart-genotyping-output" class="section level3">
<h3>Check format of DArT genotyping output</h3>
<p>Here the separator used by Diversity Array technology seems different from Marnin experience, the sep was a comma.</p>
<pre class="r"><code>vcf&lt;-read.table(dartvcfInput,
                stringsAsFactors = F,skip = 2, header = T, sep = &quot;\t&quot;, comment.char = &quot;&quot;)
vcf[1:10,1:30]
readCounts&lt;-read.csv(dartcountsInput, stringsAsFactors = F,header = T,skip=nskipcounts)
readCounts[1:10,1:50]</code></pre>
<pre><code>&gt; dim(vcf)
[1] 13603   372

&gt; dim(readCounts)
[1] 27206   406

&gt; colnames(readCounts)[1:50]
 [1] &quot;AlleleID&quot;                    &quot;CloneID&quot;                    
 [3] &quot;ClusterTempIndex&quot;            &quot;AlleleSequence&quot;             
 [5] &quot;TrimmedSequence&quot;             &quot;TrimmedSequence_plus_Strand&quot;
 [7] &quot;Short&quot;                       &quot;Lowcomplexity&quot;              
 [9] &quot;Chrom_Cassava_v61&quot;           &quot;ChromPos_Cassava_v61&quot;       
[11] &quot;SNP_ChromPos_Cassava_v61&quot;    &quot;AlnCnt_Cassava_v61&quot;         
[13] &quot;AlnEvalue_Cassava_v61&quot;       &quot;Strand_Cassava_v61&quot;         
[15] &quot;SeqDiff_Cassava_v61&quot;         &quot;ClusterConsensusSequence&quot;   
[17] &quot;ClusterSize&quot;                 &quot;AlleleSeqDist&quot;              
[19] &quot;SNP&quot;                         &quot;SnpPosition&quot;                
[21] &quot;CallRate&quot;                    &quot;OneRatioRef&quot;                
[23] &quot;OneRatioSnp&quot;                 &quot;FreqHomRef&quot;                 
[25] &quot;FreqHomSnp&quot;                  &quot;FreqHets&quot;                   
[27] &quot;PICRef&quot;                      &quot;PICSnp&quot;                     
[29] &quot;AvgPIC&quot;                      &quot;AvgCountRef&quot;                
[31] &quot;AvgCountSnp&quot;                 &quot;RatioAvgCountRefAvgCountSnp&quot;
[33] &quot;FreqHetsMinusFreqMinHom&quot;     &quot;AlleleCountsCorrelation&quot;    
[35] &quot;aggregateTagsTotal&quot;          &quot;DerivedCorrMinusSeedCorr&quot;   
[37] &quot;RepRef&quot;                      &quot;RepSNP&quot;                     
[39] &quot;RepAvg&quot;                      &quot;PicRepRef&quot;                  
[41] &quot;PicRepSNP&quot;                   &quot;TotalPicRepRefTest&quot;         
[43] &quot;TotalPicRepSnpTest&quot;          &quot;DS120xT14_17&quot;               
[45] &quot;DS120xT14_18&quot;                &quot;DS120xT14_19&quot;               
[47] &quot;DS120xT14_20&quot;                &quot;DS120xT14_21&quot;               
[49] &quot;DS120xT14_22&quot;                &quot;DS120xT14_23&quot;  

&gt; colnames(vcf)
 [1] &quot;X.CHROM&quot;     &quot;POS&quot;         &quot;ID&quot;          &quot;REF&quot;         &quot;ALT&quot;         &quot;QUAL&quot;        &quot;FILTER&quot;      &quot;INFO&quot;       
 [9] &quot;FORMAT&quot;      &quot;DS120xT14_1&quot;</code></pre>
</div>
<div id="run-conversion-function" class="section level3">
<h3>Run conversion function</h3>
<pre class="r"><code>genomicMateSelectR::convertDart2vcf(dartvcfInput, dartcountsInput, outName,
                                    nskipvcf = nskipvcf, nskipcounts = nskipcounts, ncores = 22)</code></pre>
</div>
<div id="genomewide-to-per-chrom-vcfs" class="section level3">
<h3>Genomewide to per-Chrom VCFs</h3>
<p>Split the genome-wide VCF into per-chromosome VCFs for imputation.</p>
<pre class="r"><code>require(furrr); plan(multisession, workers = 18)
options(future.globals.maxSize=+Inf); options(future.rng.onMisuse=&quot;ignore&quot;)
  
vcfIn&lt;-here::here(&quot;output&quot;,&quot;DCas21_6651.vcf.gz&quot;)
filters&lt;-&quot;--minDP 4 --maxDP 50&quot; # because using GT not PL for impute (Beagle5)
outPath&lt;-here::here(&quot;output/Report-DCas21-6651/&quot;)
outSuffix&lt;-&quot;DCas21_6651&quot;

future_map(1:18,
           ~genomicMateSelectR::splitVCFbyChr(Chr=.,
                                              vcfIn=vcfIn,filters=filters,
                                              outPath=outPath,
                                              outSuffix=outSuffix))
plan(sequential)</code></pre>
</div>
<div id="imputation-with-west-africa-refpanel" class="section level3">
<h3>Imputation with West Africa RefPanel</h3>
<p>Imputation using Beagle V5.0.</p>
<pre class="r"><code>targetVCFpath&lt;-here::here(&quot;data/Report-DCas21-6651/&quot;) # location of the targetVCF
refVCFpath&lt;-here::here(&quot;data/&quot;)
mapPath&lt;-here::here(&quot;data/CassavaGeneticMap/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
outSuffix&lt;-&quot;DCas21_6651&quot;

purrr::map(1:18,~runBeagle5(targetVCF=paste0(targetVCFpath,&quot;chr&quot;,.,
                                             &quot;_DCas21_6651.vcf.gz&quot;),
                            refVCF=paste0(refVCFpath,&quot;chr&quot;,.,&quot;_ImputationReferencePanel_StageIIpartI_72219.vcf.gz&quot;),
                            mapFile=paste0(mapPath,&quot;chr&quot;,.,&quot;_cassava_cM_pred.v6_91019.map&quot;),
                            outName=paste0(outPath,&quot;chr&quot;,.,&quot;_DCas21_6651_WA_REFimputed&quot;),
                            nthreads=112))</code></pre>
<pre class="bash"><code>mkdir BeagleLogs
cp *_DCas21_6651_WA_REFimputed.log BeagleLogs/
cp -r BeagleLogs /home/jj332_cas/lbraatz/DCas21_6651/output/
cp *_DCas21_6651_WA_REFimputed* /home/jj332_cas/lbraatz/DCas21_6651/output/
</code></pre>
</div>
<div id="post-impute-filter" class="section level3">
<h3>Post-impute filter</h3>
<pre class="r"><code>inPath&lt;-here::here(&quot;output/&quot;)
outPath&lt;-here::here(&quot;output/&quot;)
source(here::here(&quot;code&quot;,&quot;imputationFunctions.R&quot;))
require(furrr); options(mc.cores=18); plan(multiprocess)
future_map(1:18,~postImputeFilter(inPath=inPath,
                                  inName=paste0(&quot;chr&quot;,.,&quot;_DCas21_6651_WA_REFimputed&quot;),
                                  outPath=outPath,
                                  outName=paste0(&quot;chr&quot;,.,&quot;_DCas21_6651_WA_REFimputedAndFiltered&quot;)))</code></pre>
<p>Let’s check what is left</p>
<pre class="r"><code>purrr::map(1:18,~system(paste0(&quot;zcat &quot;,here::here(&quot;output/&quot;),&quot;chr&quot;,.,&quot;_DCas21_6651_WA_REFimputedAndFiltered.vcf.gz | wc -l&quot;)))</code></pre>
<pre><code>2216
1826
1915
1294
1612
1526
1031
1301
1820
924
1530
1435
1388
2337
1849
1481
975
1211</code></pre>
</div>
<div id="convert-vcf-to-dosage-matrix" class="section level3">
<h3>Convert VCF to dosage matrix</h3>
<pre class="r"><code>library(tidyverse); library(magrittr); library(genomicMateSelectR)
require(furrr); options(mc.cores=18); plan(multiprocess)
pathOut&lt;-here::here(&quot;output/&quot;)

# DCas21_6651
future_map(1:18,~convertVCFtoDosage(pathIn=here::here(&quot;output/&quot;),pathOut=pathOut,
                                    vcfName = paste0(&quot;chr&quot;,.,&quot;_DCas21_6651_WA_REFimputedAndFiltered&quot;)))
# Genome-wide dosage (for use in R) for each dataset
# DCas21_6651
createGenomewideDosage(pathIn = here::here(&quot;output/&quot;), chroms=1:18, &quot;_DCas21_6651_WA_REFimputedAndFiltered&quot;)</code></pre>
<pre class="bash"><code>cp *_DCas21_6651_WA_REFimputedAndFiltered.* /home/jj332_cas/lbraatz/DCas21_6651/output/.
cp DosageMatrix_DCas21_6651_WA_REFimputedAndFiltered.rds /home/jj332_cas/lbraatz/DCas21_6651/output/.
cp DCas21_6651.* /home/jj332_cas/lbraatz/DCas21_6651/output/.</code></pre>
</div>
<div id="test-markers-to-see-their-polymorphism" class="section level3">
<h3>Test Markers to see their polymorphism</h3>
<pre class="r"><code>suppressMessages(library(tidyverse)); library(here)
IITAMrks &lt;- readRDS(here::here(&quot;output&quot;, &quot;DosageMatrix_DCas21_6651_WA_REFimputedAndFiltered.rds&quot;))

freq &lt;- colMeans(IITAMrks)/2
sum(freq &gt; 0.5)

for(i in 1:length(freq)){
  if(freq[i] &gt; 0.5){
    freq[i] &lt;- (1 - freq[i])
    }
}

sum(freq &gt;= 0.05)

PIC &lt;- tibble(SNPs = names(freq),
              PIC = (2*(1-freq)*freq))

PIC %&gt;% ggplot(aes(x = PIC)) +
  geom_density()</code></pre>
</div>
<div id="do-not-run---still-working-on" class="section level3">
<h3>Do not Run - Still working on</h3>
<pre class="bash"><code>export PATH=/programs/plink-1.9-x86_64-beta3.30:$PATH 
plink --vcf &quot;, pathIn, vcfName, &quot;.vcf.gz &quot;, &quot;--make-bed --const-fid --keep-allele-order &quot;, 
        &quot;--out &quot;, pathOut, vcfName))
    system(paste0(&quot;zcat &quot;, pathIn, vcfName, &quot;.vcf.gz &quot;, &quot;| cut -f1-5 &gt; &quot;, 
        pathOut, vcfName, &quot;.sitesWithAlleles&quot;))
    read.table(paste0(pathOut, vcfName, &quot;.sitesWithAlleles&quot;), 
        stringsAsFactors = F, header = F, sep = c(&quot;\t&quot;)) %&gt;% 
        select(V3, V5) %&gt;% write.table(., file = paste0(pathOut, 
        vcfName, &quot;.alleleToCount&quot;), row.names = F, sep = c(&quot;\t&quot;), 
        quote = F, col.names = F)
    system(paste0(&quot;export PATH=/programs/plink-1.9-x86_64-beta3.30:$PATH;&quot;, 
        &quot;plink --bfile &quot;, pathOut, vcfName, &quot; --keep-allele-order --recode A &quot;, 
        &quot;--recode-allele &quot;, pathOut, vcfName, &quot;.alleleToCount &quot;, 
        &quot;--out &quot;, pathOut, vcfName))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Big Sur 11.6.5

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     rstudioapi_0.13  whisker_0.4      knitr_1.38      
 [5] magrittr_2.0.3   workflowr_1.7.0  R6_2.5.1         rlang_1.0.2     
 [9] fastmap_1.1.0    fansi_1.0.3      stringr_1.4.0    tools_4.1.2     
[13] xfun_0.30        utf8_1.2.2       cli_3.2.0        git2r_0.30.1    
[17] jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2   rprojroot_2.0.3 
[21] yaml_2.3.5       digest_0.6.29    tibble_3.1.6     lifecycle_1.0.1 
[25] crayon_1.5.1     later_1.3.0      sass_0.4.1       vctrs_0.4.0     
[29] promises_1.2.0.1 fs_1.5.2         glue_1.6.2       evaluate_0.15   
[33] rmarkdown_2.13   stringi_1.7.6    bslib_0.3.1      compiler_4.1.2  
[37] pillar_1.7.0     jsonlite_1.8.0   httpuv_1.6.5     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
